src/plone/restapi/services/blocktypes/get.py

```diff
@@ -0,0 +1,39 @@
+from plone import api
+from plone.restapi.services import Service
+from collections import Counter
+from plone.dexterity.content import get_assignable
+
+
+class BlockTypesGet(Service):
+    def reply(self):
+        catalog = api.portal.get_tool(name="portal_catalog")
+        request_body = self.request.form
+        result = {}
+
+        if request_body.get("blocktypes") != "":
+            blocktypes = request_body.get("blocktypes").split(",")
+
+            for blocktype in blocktypes:
+                brains = catalog.searchResults(block_types=blocktype)
+                result[blocktype] = Counter()
+
+                for brain in brains:
+                    obj = brain.getObject()
```

Comment:
The above query could ask the `object_provides` index for the (marker) interface of the `volto.blocks` behavior. This would several an object wake ups.

Reply:
I just tried the `object_provides` index for the marker interface of the `volto.blocks` behavior like so: `brains = catalog(object_provides=IBlocks.__identifier__)` which returns brains just like the `block_types` index. And so I am asking my self: Doesn't this make the `block_types` index completely unnecessary? Because from the brains I get the object and from there I get the block values.

/cc @davisagli

Reply:
@jonaspiterek You can do that but then you end up loading the full object for all the content items that have blocks, and loading the full object is slower than loading the metadata that is stored as part of the catalog. The block_types index exists for the use case where you want to find all the items that have a particular type of block, without actually loading the full objects.

Also the block_types index [properly finds blocks that are nested inside other blocks](https://github.com/plone/plone.volto/blob/main/src/plone/volto/indexers.py#L44), which your code doesn't do yet.

Reply:
@jonaspiterek Btw if you just want to find out how many items have each type of block, you can also get that efficiently from the index like this (if I recall correctly): `api.portal.get_tool("portal_catalog").Indexes["block_types"].uniqueValues(withLengths=1)`